lib/puppet/parser/functions/validate_integer.rb

```diff
@@ -53,6 +53,8 @@ module Puppet::Parser::Functions
 
     ENDHEREDOC
 
+    function_deprecation([:validate_integer, 'This method is deprecated, please use the stdlib validate_legacy function, with Stdlib::Compat::Numeric. There is further documentation for validate_legacy function the README.'])
```

Comment:
1. should all of these deprecation use the same key?
2. all of these deprecations should point to the right type (not ::Numeric, when it should be ::Integer)
3. "There is further documentation for validate_legacy function **>>in<<** the README"